src/builder.rs

```diff
@@ -661,7 +661,7 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
         // dereference after a drop, for instance.
         // TODO(antoyo): handle align of the load instruction.
         let deref = ptr.dereference(None).to_rvalue();
-        let value_type = deref.get_type();
+        let value_type = deref.get_type().unqualified();
```

Comment:
Why was this added?

Reply:
This strips `const` qualification to make the defined local variable writable. Writing to a `const` local variable causes a libgccjit error (which is presumably due to [`memento_of_get_const::accepts_writes_from`](https://github.com/antoyo/gcc/blob/141e034e736dd375a1236d5bc0a5b9ba0c6a274c/gcc/jit/jit-recording.h#L757-L761) returning `false`):

```
[BUILD] mini_core
   ⋮
libgccjit.so: error: gcc_jit_block_add_assignment: mismatching types: assignment to loadedValue50 (type: const __uint8_t) from *&_RNvNvCs2VSKAzIc0pl_9mini_core7get_tls1A (type: const __uint8_t)
```

Reply:
Hum, that could possibly remove other important qualifiers (like volatile, aligned).
I'll write a patch to libgccjit to have the possibility to only remove const.

Reply:
Actually, that might not be necessary: `volatile` never appears in this place. The alignment is explicitly specified by the caller (`align: Align`), so we just have to apply it by `get_aligned`.

Reply:
Never mind, `value_type` should have the natural alignment of the pointed type.

Reply:
With your edits, I'm not sure what's needed now.

Do you wait for my patch to only remove const?

Reply:
Oh, sorry for leaving you hanging. Yes, I'm awaiting your patch.

Reply:
I added it [here](https://github.com/antoyo/gcc/commit/72cb89f82a6c9c5471f0b2c122b21dbe387c7904) and [there](https://github.com/antoyo/gccjit.rs/commit/383523c29c9dde572d2079ad21decb572da24cc9).